wgsl/index.bs

```diff
@@ -3018,7 +3040,7 @@ short_circuit_or_expression
 
 <pre class='def'>
 assignment_statement
-  : singular_expression EQUAL short_circuit_or_expression
+  : STAR singular_expression EQUAL short_circuit_or_expression
```

Comment:
Is it intentional that all variables be treated as pointers? It would be worth adding that to the pointer operations examples and calling out in the variables section.

Reply:
Maybe accidental but that's the direction I'd prefer.